firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/RemoteSettings.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions.settings
+
+import android.content.Context
+import android.net.Uri
+import androidx.datastore.preferences.preferencesDataStore
+import java.net.URL
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+internal class RemoteSettings(val context: Context) : SettingsProvider {
+  private val SESSION_CONFIGS_NAME = "firebase_session_settings"
+  private val Context.dataStore by preferencesDataStore(name = SESSION_CONFIGS_NAME)
```

Comment:
Is this adding an extension property to the Context class? Is the scope limited to this instance of this class?

Reply:
Yes to both the questions. The key reason is that we should not create more than once instance of the preference data store and the the recommendation is to add it as a extension to the Context class. But I want to limit the exposure of this dataStore to just this class.